### ISSUE TITLE
Support flow type props validation

### ISSUE THREAD

**Issue by ndelangen** [storybooks/babel-plugin-react-docgen#29]
Did you have any plan to support:

``` js
export default class Item extends React.Component { // eslint-disable-line
  props: {
    icon: string,
    tooltip?: string,
    onClick?: () => void,
  };

  render() {
    const { icon, tooltip, onClick } = this.props
    return (
      <span className={styles.root} onMouseDown={onClick} title={tooltip}>
        <span className="material-icons">{icon}</span>
      </span>
    )
  }
}

```

together with

``` js
  static propTypes = {
    icon: React.PropTypes.string.isRequired,
    tooltip: React.PropTypes.string,
    onClick: React.PropTypes.func,
  };
```

**Comment by danielduan** [comment]
I'm not actually sure how flow works under the hood. If there's a way to get flow type transpiled in babel to `static propTypes` before we process the docgen stuff, there's definitely possibility.

**Comment by danielduan** [comment]
Gonna close this one since it seems like react-docgen has support for flow-types in the latest versions:
https://github.com/reactjs/react-docgen

**Comment by Mottoweb** [comment]
It seems that info addon still does not render props from flow types though.

**Comment by danielduan** [comment]
@Mottoweb try cleaning out your `node_modules` and reinstalling. You'll need the latest version of info and the associated babel plugin.

**Comment by alexeyMohnatkin** [comment]
Should it work with flow types only?

**Comment by danielduan** [comment]
I'm not sure what you mean by flow types only @alexeyMohnatkin. This should work with both regular prop types and flow types.

**Comment by Mottoweb** [comment]
@danielduan it worked thanks, but still i cant see the description for the prop. Or maybe its addon-info related?

**Comment by danielduan** [comment]
@Mottoweb there's been a bug fix merged to 3.3 that fixes the docgen issues when loading from static bundles.
